Set-DomainUser.ps1

```diff
@@ -1,720 +1,1124 @@
-# bnRlYyBTSEEyNTYgVGltZVN0YW1waW5nIENBAhB71OWvuswHP6EBIwQiQU0SMAsG
-# CWCGSAFlAwQCAaCBpDAaBgkqhkiG9w0BCQMxDQYLKoZIhvcNAQkQAQQwHAYJKoZI
-# hvcNAQkFMQ8XDTIyMDQyNzEzNTQyN1owLwYJKoZIhvcNAQkEMSIEIKdj25w7Hqsh
-# 4uhBWbC5e6K74b1PmmvKasnqqQiUxHH1MDcGCyqGSIb3DQEJEAIvMSgwJjAkMCIE
-# IMR0znYAfQI5Tg2l5N58FMaA+eKCATz+9lPvXbcf32H4MAsGCSqGSIb3DQEBAQSC
-# AQCfI+x4vYtiXEBmJSNgN3vTbmR0SJvpnpRcyc26RaMYVnEut58sgu760U1XLuFK
-# yeVr3BwqAiONAYG6XfaB4CMCrJL8WVT2EAC0GA4s5rKolIqL8zbigNcmPSUOJuKM
-# biOPVpuBELblSOz+bTIbHrOceioetkBrtv6+nsHEKRmkZzysGfqKYDuSd0S5MccL
-# HDh+9alc9nG8CGfMitvVIoNlxh8/t/Luv0GOCDo/NBYYfKaQ+RyEjEgFM0kqLagE
-# zR7UFJHpNj04CcjxsVPckhXlfYGumh0/VFANH7pC+ziBggngctuXyneLCt5XztIY
-# vXOpmdYDFO3VL+S3EQ+EVSPm
-# SIG # End signature block
+### Script to help move the PSM users from local to domain users
+
+<#
+.SYNOPSIS
+This script will update the connector server to a domain user setup. It will also onboard the domain users into the portal inside the PSM safe.
+.DESCRIPTION
+Does the Domain User for PSM setup.
+.PARAMETER pvwaAddress
+The PVWA Address (https://tenant.privilegecloud.cyberark.com, or on-prem URL)
+.PARAMETER domain
+The domain of the domain user account(s).
+.PARAMETER NETBIOS
+The NETBIOS for the domain user account(s).
+.PARAMETER safe
+The safe in which to store PSM user credentials
+.PARAMETER tinaCreds
+Tenant Administrator/InstallerUser credentials
+.PARAMETER psmConnectCredentials
+PSMConnect domain user credentials
+.PARAMETER psmAdminCredentials
+PSMAdminConnect domain user credentials
+.PARAMETER TestPsmConnectCredentials
+Validate psmConnectCredentials domain user credentials
+.PARAMETER TestPsmAdminCredentials
+Validate PSMAdminConnect domain user credentials
+.PARAMETER IgnoreShadowPermissionErrors
+Ignore errors while granting PSMAdminConnect user shadow permissions
+.PARAMETER PlatformName
+The name of the platform to be created for the PSM accounts
+.PARAMETER PSMConnectAccountName
+The Account Name for the object in the vault which will contain the PSMConnect account. Defaults to "PSMConnect".
+.PARAMETER PSMAdminConnectAccountName
+The Account Name for the object in the vault which will contain the PSMAdminConnect account. Defaults to "PSMAdminConnect".
+.PARAMETER DoNotHarden
+Skip running the PSMHardening.ps1 script to speed up execution if step has already been completed.
+.PARAMETER DoNotConfigureAppLocker
+Skip running the PSMConfigureAppLocker.ps1 script to speed up execution if step has already been completed.
+#>
+
+
+
+[CmdletBinding()]
+param(
+    [Parameter(
+        Mandatory = $false,
+        HelpMessage = "Please enter the full PVWA Address IE: https://tenantname.privilegecloud.cyberark.cloud")]
+    [string]$pvwaAddress,
+    [Parameter(
+        Mandatory = $false,
+        HelpMessage = "Please enter the domain of the created accounts IE: lab.net")]
+    [string]$domain,
+    [Parameter(
+        Mandatory = $false,
+        HelpMessage = "Please enter the NETBIOS of the created accounts IE: LAB")]
```

Comment:
These parameters are optional now because we'll try to autodetect them. If this fails we'll ask for them to be added on the command line.